yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher.java

```diff
@@ -75,12 +86,55 @@ public JNLPLauncher getJnlpLauncher() {
     }
 
     @DataBoundSetter
-    public void setUser(String user) {
-        this.user = user;
+    public void setSlaveOpts(@Nonnull String slaveOpts) {
+        this.slaveOpts = StringUtils.trimToEmpty(slaveOpts);
+    }
+
+    public String getSlaveOpts() {
+        return StringUtils.trimToEmpty(slaveOpts);
+    }
+
+    @DataBoundSetter
+    public void setJenkinsUrl(@Nonnull String jenkinsUrl) {
+        this.jenkinsUrl = StringUtils.trimToEmpty(jenkinsUrl);
+    }
+
+    public String getJenkinsUrl() {
+        return StringUtils.trimToEmpty(jenkinsUrl);
+    }
+
+    public String getJenkinsUrl(@Nonnull String rootUrl) {
+        if (!isNull(jenkinsUrl) && StringUtils.isNotEmpty(jenkinsUrl)) {
+            return jenkinsUrl;
+        }
+        return StringUtils.trimToEmpty(rootUrl);
+    }
+
+    @DataBoundSetter
+    public void setJvmOpts(@Nonnull String jvmOpts) {
+        this.jvmOpts = StringUtils.trimToEmpty(jvmOpts);
+    }
+
+    public String getJvmOpts() {
```

Comment:
hummm... sorry, but it is still not clear to me. If the function returns from trimToEmpty, why testing Nonnull?